### PR TITLE
Feature/error handling

### PR DESCRIPTION
**Pull request by masadmunawar**
Adds error Handling on Backend side via JSON file

### CODE REVIEW

File: src/routers/daignonsisResult.js [hunk 1]
```diff
@@ -85,36 +102,54 @@ router.get("/calculateDiagnotics/:countryCode", async (req, res) => {
...
           possibleDiagnostic: populateDiagnosis[i].diagnosticsId.diagnosisName,
-          percentage: percentage
+          percentage: percentage,
         };
-      }
-      else {
-        res.status(400).json({ success: `"${req.params.countryCode}" this countryCode is not available` });
+      } else {
+        // If the country code is not "es" or "en", retrieve the error message for invalid country code
+        const errorMessage = errorMessageEs.INVALID_COUNTRY_CODE;
```
Comment (anees-rehman-techtimize) [inline src/routers/daignonsisResult.js:129]: Here, how do you know that country code is "ES" and you are getting the error message in Spanish?

File: src/routers/daignonsisResult.js [hunk 2]
```diff
@@ -85,36 +102,54 @@ router.get("/calculateDiagnotics/:countryCode", async (req, res) => {
...
       resultPercentage.push(diagnosisObj);
     }
     // Send the resultPercentage array as a response with a status code of 200 (OK)
     res.status(200).send(resultPercentage);
   } catch (err) {
-    // Respond with a status code of 500 and the error message if there was an issue retrieving the data
-    res.status(500).send(err);
+    // If an error occurs, retrieve the error message for failed calculate the diagnosis result
+    const errorMessage =
+      req.params.countryCode === "es"
```
Comment (anees-rehman-techtimize) [inline src/routers/daignonsisResult.js:143]: Check country code from Enum file

File: src/routers/daignonsisResult.js (hunk 1 above)
Comment (anees-rehman-techtimize) [reply src/routers/daignonsisResult.js:129]: Return EN message by default.

File: src/routers/daignonsisResult.js [hunk 3]
```diff
@@ -85,36 +102,54 @@ router.get("/calculateDiagnotics/:countryCode", async (req, res) => {
...
         diagnosisObj = {
-          possibleDiagnostic: populateDiagnosis[i].diagnosticsId.diagnosisNameEs,
-          percentage: percentage
+          possibleDiagnostic:
+            populateDiagnosis[i].diagnosticsId.diagnosisNameEs,
+          percentage: percentage,
         };
-      }
-      else if (req.params.countryCode == CountryCode.ENGLISH) {
+      } else if (req.params.countryCode == CountryCode.ENGLISH) {
```
Comment (anees-rehman-techtimize) [inline src/routers/daignonsisResult.js:122]: Add a new country code "en-us" and before checking the code, convert it to lowercase.

File: src/routers/daignonsisResult.js [hunk 4]
```diff
@@ -6,55 +6,68 @@ const Probability = require("../models/probability");
...
   // Find the pain behavior question based on the painBehaviorId
-  const result = PainBehaviorQuestion.find({ painBehaviorId: request.painBehaviorId }).populate("questionId");
+  const result = PainBehaviorQuestion.find({
+    painBehaviorId: request.painBehaviorId,
+  }).populate("questionId");
   return result;
-}
+};
 
 router.get("/calculateDiagnotics/:countryCode", async (req, res) => {
```
Comment (anees-rehman-techtimize) [inline src/routers/daignonsisResult.js:72]: it should be POST request

File: src/routers/daignonsisResult.js [hunk 5]
```diff
@@ -85,36 +102,54 @@ router.get("/calculateDiagnotics/:countryCode", async (req, res) => {
...
+        // Return the error message with a status code of 400 Bad Request
+        res.status(errorMessage.statusCode).send({
+          success: false,
+          message: `${errorMessage.message}: "${req.params.countryCode}"`,
+        });
       }
       resultPercentage.push(diagnosisObj);
     }
     // Send the resultPercentage array as a response with a status code of 200 (OK)
     res.status(200).send(resultPercentage);
```
Comment (anees-rehman-techtimize) [inline src/routers/daignonsisResult.js:139]: Return only top 3 diagnosis in ascending order of percentage.

**Review by anees-rehman-techtimize — CHANGES_REQUESTED** [review]
please make the required changes

File: src/routers/daignonsisResult.js (hunk 1 above)
Comment (masadmunawar) [reply src/routers/daignonsisResult.js:129]: done

File: src/routers/daignonsisResult.js (hunk 3 above)
Comment (masadmunawar) [reply src/routers/daignonsisResult.js:122]: done

File: src/routers/daignonsisResult.js (hunk 2 above)
Comment (masadmunawar) [reply src/routers/daignonsisResult.js:143]: done

File: src/routers/daignonsisResult.js (hunk 4 above)
Comment (masadmunawar) [reply src/routers/daignonsisResult.js:72]: done

**Review by anees-rehman-techtimize — APPROVED** [review]

**Pull request merged**